### PR TITLE
Adding gigachat LLM support

### PR DESCRIPTION
**Pull request by Afaneor**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

I created a new integration package for [gigachat](https://developers.sber.ru/docs/ru/gigachat/api/overview) llm.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

I have unittests for every method in class, and tested without mocks in my personal project.

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-gigachat/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-gigachat/poetry.lock:1]: no need to commit the lock file

File: llama-index-integrations/llms/llama-index-llms-gigachat/README.md
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-gigachat/README.md:1]: This readme shows up on https://llamahub.ai, so we can add a few more details (installation, how to use, etc.)

File: llama-index-integrations/llms/llama-index-llms-gigachat/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Feel free to un-skip examples, and experimental, you will just need to
+# work through many typos (--write-changes and --interactive will help)
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.llms.gigachat"
+
+[tool.llamahub.class_authors]
+CLASS = "afaneor"
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-gigachat/pyproject.toml:17]: lets put the class name here

File: llama-index-integrations/llms/llama-index-llms-gigachat/llama_index/llms/gigachat/base.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                )
+            )
+        return ChatResponse(
+            message=ChatMessage(
+                content=response.choices[0].message.content,
+                role="assistant",
+            ),
+        )
+
+    # @llm_completion_callback()
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-gigachat/llama_index/llms/gigachat/base.py:170]: any reason to leave this commented out?

File: llama-index-integrations/llms/llama-index-llms-gigachat/poetry.lock
Comment (Afaneor) [reply llama-index-integrations/llms/llama-index-llms-gigachat/poetry.lock:1]: removed lock file

File: llama-index-integrations/llms/llama-index-llms-gigachat/README.md
Comment (Afaneor) [reply llama-index-integrations/llms/llama-index-llms-gigachat/README.md:1]: Added examples of usage in readme

File: llama-index-integrations/llms/llama-index-llms-gigachat/pyproject.toml (hunk 1 above)
Comment (Afaneor) [reply llama-index-integrations/llms/llama-index-llms-gigachat/pyproject.toml]: Fixed, added

File: llama-index-integrations/llms/llama-index-llms-gigachat/llama_index/llms/gigachat/base.py (hunk 2 above)
Comment (Afaneor) [reply llama-index-integrations/llms/llama-index-llms-gigachat/llama_index/llms/gigachat/base.py]: Refactored methods, now decorator is uncommented and it works

**Comment by Afaneor** [comment]
@logan-markewich, fixed your comments in new commits

**Review by logan-markewich — APPROVED** [review]

**Comment by Afaneor** [comment]
@logan-markewich, it looks like pipeline failed due to CI/CD configuration, nltk related.

Can i fix it by myself?

**Comment by Afaneor** [comment]
@logan-markewich, it looks like pipeline failed due to CI/CD configuration, nltk related.

Can i fix it by mysself

**Comment by logan-markewich** [comment]
@Afaneor I fixed the nltk issue, looks like the actual tests are broken though

**Comment by Afaneor** [comment]
fixed tests in last commit

**Comment by Afaneor** [comment]
@logan-markewic, looks like the package is not published to pypi, how can i fix it?

**Pull request merged**